File: 贪心算法/55.跳跃游戏.java
class Solution {
    public boolean canJump(int[] nums) {
        int max=nums[0];
        for(int i=0;i<=max;i++){
            max=Math.max(max,i+nums[i]);
            if(max>=nums.length-1){
                return true;
            }
        }
        return false;
    }
}

File: 贪心算法/55.跳跃游戏.go
func canJump(nums []int) bool {
    n:=len(nums)-1
    res:=0
    for i:=0;i<=res;i++{
        res=max(res,i+nums[i])
        if res>=n{
            return true
        }
    }
    return false
}